User can tap Login button on Launch screen

andela-cofor opened the pull request:
#### What does this PR do?
adds login button 

#### Description of Task to be completed?
When I click the login button, i am navigated to the Login screen
Functionality is properly tested. i.e Test coverage for related code > 85%

#### How should this be manually tested?
N/A

#### Any background context you want to provide?
N/A

#### Screenshots 
<img width="370" alt="screen shot 2018-01-27 at 4 28 17 pm" src="https://user-images.githubusercontent.com/25608370/35473327-21846006-037f-11e8-9482-348b103ce393.png">


#### Questions:
N/A

andela-cofor committed:
Create README.md

andela-cofor committed:
add login button

Login button navigates to login screen

andela-cofor committed:
start test

oofoamalu committed:
Update gitignore file

oofoamalu committed:
Update gitignore file

oofoamalu commented on .gitignore:
```diff
@@ -7,5 +7,6 @@
 /build
 /captures
 .externalNativeBuild
+/.idea/misc.xml
```
This is a duplicate. You can remove this line

andela-cofor replied:
Ok, I have removed it.

oofoamalu commented on app/build.gradle:
```diff
@@ -1,6 +1,9 @@
 
 android {
+    testOptions {
+        unitTests.returnDefaultValues = true
```
Can you enlighten me on what this configuration does?

andela-cofor replied:
I was trying to get my tests to run but it did not, I have taken it out thanks for pointing that out.

oofoamalu commented on .circleci/config.yml:
```diff
@@ -0,0 +1,42 @@
+# Java Gradle CircleCI 2.0 configuration file
```
Revert this. We shouldn't this update on this PR

andela-cofor replied:
I see, I don't have that file locally, it appears only on git hub so I assume circleci updated the PR when we installed it, that's why I told you tests are failing...

oofoamalu commented on .gitignore:
```diff
@@ -6,4 +6,3 @@
 /build
 /captures
 .externalNativeBuild
-/.idea/modules.xml
```
I'm not sure why you deleted this.

andela-cofor replied:
Duplicate

oofoamalu committed:
Remove duplicate

andela-cofor committed:
delete unused code from gradle file

oofoamalu commented on app/src/androidTest/java/com/blackboxteam/app/blackbox/LaunchScreenTest.java:
```diff
@@ -0,0 +1,38 @@
+            LaunchScreenActivity.class);
+
+    @Test
+    public void navigateToLogin() {
```
This test is incomplete. The standard format is:
- Perform an action
- Verify an outcome/output

Here, you have only performed an action. You would need to verify the outcome of that action.

oofoamalu commented on app/src/main/java/com/blackboxteam/app/blackbox/LaunchScreenActivity.java:
```diff
@@ -14,21 +14,21 @@
 public class LaunchScreenActivity extends AppCompatActivity implements View.OnClickListener {
 
-    @BindView(R.id.secureButton) Button nextScreen;
+    @BindView(R.id.launchScreenLoginButton) Button mloginScreen;
```
As a style guide, the annotation should be on top the variable. Move the button variable to the next line.

oofoamalu commented on app/src/main/java/com/blackboxteam/app/blackbox/LaunchScreenActivity.java:
```diff
@@ -14,21 +14,21 @@
     }
 
     @Override
     public void onClick(View v) {
```
There is a better way to set click listeners. Remove the activity level click listener declaration.

oofoamalu commented on app/src/main/res/values/colors.xml:
```diff
@@ -4,8 +4,11 @@
 
-    <color name="secureButtonBackgroundColor">#5c5c5c</color>
-    <color name="secureButtonTextColor">#fff</color>
+    <color name="buttonBackgroundColor">#5c5c5c</color>
```
Renaming this variable caused additional changes to files that are not relevant to this PR. Find a cleaner way to define this colour variable.

oofoamalu commented on app/src/main/res/values/colors.xml:
```diff
@@ -4,8 +4,11 @@
-    <color name="secureButtonBackgroundColor">#5c5c5c</color>
-    <color name="secureButtonTextColor">#fff</color>
+    <color name="buttonBackgroundColor">#5c5c5c</color>
+    <color name="launchScreenButtonTextColor">#fff</color>
```
Same for this.

oofoamalu commented on app/src/androidTest/java/com/blackboxteam/app/blackbox/LaunchScreenTest.java:
```diff
@@ -0,0 +1,38 @@
+import static junit.framework.Assert.assertNotNull;
+
+/**
+ * Created by oforchinedu on 1/29/18.
```
Java doc should be descriptive. Update comment to explain what the class does.

oofoamalu reviewed (changes requested):
Good work overrall. Please address all PR comments.

andela-cofor commented:
WIP

eshaibu committed:
adjust naming convention for colors

eshaibu committed:
use butterknife onclick functionality

eshaibu committed:
update test to assert next login screen on click

eshaibu committed:
Merge branch 'develop' of github.com:andela-cofor/blackbox into ft/login-button

# Conflicts:
#	app/build.gradle
#	build.gradle
#	gradle/wrapper/gradle-wrapper.properties

eshaibu committed:
Revert "Merge branch 'develop' of github.com:andela-cofor/blackbox into ft/login-button"

This reverts commit 33e474ee52087d92b372832934b43b91a2a5877a, reversing
changes made to 68b4b38bf7632f58db287e1dba566eb49b46ecc0.

oofoamalu reviewed (changes requested):
@andela-eshaibu It wasn't necessary to change the gradle file configurations. Undo these changes.

oofoamalu commented on app/src/main/java/com/blackboxteam/app/blackbox/LaunchScreenActivity.java:
```diff
@@ -9,29 +9,26 @@
-                startActivity(intent);
-        }
+    @OnClick(R.id.launchScreenLoginButton)
+    public void launchScreenLoginButton(){
```
Add a space before the opening brace i.e {

oofoamalu commented on gradle/wrapper/gradle-wrapper.properties:
```diff
@@ -1,6 +1,7 @@
-#Sat Feb 03 00:36:23 WAT 2018
+#Tue May 30 12:48:27 WAT 2017
```
Undo changes to this file.

oofoamalu commented on build.gradle:
```diff
@@ -7,10 +7,9 @@ buildscript {
             url 'https://maven.google.com/'
```
Undo changes to this file. We want to use the most recent build versions

oofoamalu commented on app/build.gradle:
```diff
@@ -1,12 +1,12 @@
 apply plugin: 'com.android.application'
```
Undo changes to this file. We want to use the latest versions

ckOfor committed:
update gradle and buildVersion

andela-cofor committed:
implement feedback
- modify build.gradle file to match development
- modify gradle properties file to match development
- add space to methods and class

oofoamalu commented on app/build.gradle:
```diff
@@ -23,15 +23,20 @@ android {
 dependencies {
-    implementation fileTree(dir: 'libs', include: ['*.jar'])
-    androidTestImplementation('com.android.support.test.espresso:espresso-core:2.2.2', {
+    compile fileTree(dir: 'libs', include: ['*.jar'])
```
These should be implementation not compile

oofoamalu replied:
@andela-cofor

andela-cofor committed:
update gradle from compile to implementation
